src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewList.cs

```diff
@@ -0,0 +1,689 @@
+				{
+					// If not in multi-select mode, collapse the dragged node.
+					m_draggedTreeViewNode.IsExpanded = false;
+				}
+
+				VisualStateManager.GoToState(
+					tvItem,
+					isMultipleDragging ? "MultipleDraggingPrimary" : "Dragging",
+					false);
+				UpdateDropTargetDropEffect(false, false, null);
+			}
+		}
+
+		private void OnDragItemsCompleted(object sender, DragItemsCompletedEventArgs args)
+		{
+			m_draggedTreeViewNode = null;
+			m_emptySlotIndex = -1;
+		}
+
+		private void OnContainerContentChanging(object sender, ContainerContentChangingEventArgs args)
+		{
+			//	// Only need to update unrecycled on screen items.
+			if (!args.InRecycleQueue)
+			{
+				var targetItem = (TreeViewItem)args.ItemContainer;
+				var targetNode = NodeFromContainer(targetItem);
+				var treeViewItem = targetItem;
+				treeViewItem.UpdateIndentation(targetNode.Depth);
+				treeViewItem.UpdateSelectionVisual(targetNode.SelectionState);
+			}
+		}
+
+		// IControlOverrides
+		protected override void OnDrop(DragEventArgs e)
+		{
+			DragEventArgs args = e;
+
+			if (args.AcceptedOperation == DataPackageOperation.Move && !args.Handled)
+			{
+				if (m_draggedTreeViewNode != null && IsIndexValid(m_emptySlotIndex))
+				{
+					// Get the node at which we will insert
+					TreeViewNode insertAtNode = NodeAtFlatIndex(m_emptySlotIndex);
+
+					if (IsMutiSelectWithSelectedItems())
+					{
+						// Multiselect drag and drop. In the selected items, find all the selected subtrees 
+						// and move each of those subtrees.
+						var selectedRootNodes = GetRootsOfSelectedSubtrees();
+						var selectionSize = selectedRootNodes.Count;
+
+						// Loop through in reverse order because we are inserting above the previous item to get the order correct.
+						for (int i = selectedRootNodes.Count - 1; i >= 0; --i)
+						{
+							MoveNodeInto(selectedRootNodes[i], insertAtNode);
+						}
+					}
+					else
+					{
+						MoveNodeInto(m_draggedTreeViewNode, insertAtNode);
+					}
+
+					UpdateDropTargetDropEffect(false, false, null);
+					args.Handled = true;
+				}
+			}
+
+			base.OnDrop(e);
+		}
+
+		private void MoveNodeInto(TreeViewNode node, TreeViewNode insertAtNode)
+		{
+			int nodeFlatIndex = FlatIndex(node);
+			if (insertAtNode != node && IsFlatIndexValid(nodeFlatIndex))
+			{
+				RemoveNodeFromParent(node);
+
+				int insertOffset = nodeFlatIndex < m_emptySlotIndex ? 1 : 0;
+				// if the insertAtNode is a parent that is expanded
+				// insert as the first child
+				if (insertAtNode.IsExpanded && insertOffset == 1)
+				{
+					insertAtNode.Children.Insert(0, node);
+				}
+				else
+				{
+					// Add the item to the new parent (parent of the insertAtNode)
+					var children = insertAtNode.Parent.Children;
+					int insertNodeIndexInParent = IndexInParent(insertAtNode);
+					children.Insert(insertNodeIndexInParent + insertOffset, node);
+				}
+			}
+		}
+
+		protected override void OnDragOver(DragEventArgs args)
+		{
+			if (!args.Handled)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'args' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)